specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1136 @@
+## 3.5. Reset
+
+`PackageDeploymentManager` offers Reset APIs providing the same functionality as available
+interactively via Settings' `Reset` button on the detail page for an app (via Apps > Installed Apps
+> ... > Advanced options).
+
+## 3.6. IsPackageRegistrationPending
+
+`IsPackageRegistrationPending()` detects if package registration is pending for the specified target. For
+example, if a package is in use and `AddPackageByUriAsync()` is called with a newer version and the
+option
+[DeferRegistrationWhenPackagesAreInUse](https://learn.microsoft.com/uwp/api/windows.management.deployment.addpackageoptions.deferregistrationwhenpackagesareinuse)=`true`
+then the registration is delayed until the package is no longer in use and can be updated. In such
+cases `IsPackageRegistrationPending()` returns `true`.
+
+##  3.7. PackageSets
+
+A `PackageSet` is a group of packages to be operated on with one request. Package sets provide a
+convenient means to perform multiple operations.
+
+For example, `IsPackageSetReady(ps)` returns `true` only if all packages referenced by the
+`PackageSet` are ready for use. For another example, `AddPackageSetAsync(ps, options)` is
+functionally equivalent to
+
+```c#
+var pdm = PackageDeploymentManager().GetDefault();
+foreach (PackageSetItem psi in ps.PackageSetItems)
+{
+    var result = await pdm.AddPackageAsync(psi.PackageUri, options)
+    if (result.Status != PackageDeploymentStatus.CompletedSuccess)
+    {
+        return result;
+    }
+}
+return new PackageDeploymentResult(PackageDeploymentStatus.CompletedSuccess);
+```
+
+## 3.8. PackageRuntimeManager
+
+The `PackageRuntimeManager` API provides Dynamic Dependency support for PackageSet operations,
+especially when the caller may not know the exact package(s) involved (for instance, when targeting
+packages via `ms-uup:`). `PackageRuntimeManager` determines the packages involved for a `PackageSet`
+and dynamically adds them caller's package graph.
+
+## 3.9. PackageVolume Repair
+
+Packages are typically<sup>1</sup> installed to a
+[PackageVolume](https://learn.microsoft.com/uwp/api/windows.management.deployment.packagevolume),
+e.g. `C:\Program Files\WindowsApps` is the default `PackageVolume` on a new Windows system. Windows
+pairs the PackageVolume with the underlying storage volume's media ID to identify the PackageVolume
+even when mount points change (e.g. instead USB flash key and it's mount point is E:, the remove the
+USB flash key and later reinsert it but not it's mounted as K:).
+
+It's possible this tracking information can be invalidated, e.g. backup a drive's content, replace
+the drive with a new one and then restore the content. The packages installed on this drive's
+PackageVolume(s) aren't recognized by Windows because the new drive has a different media ID.
+
+The Windows App SDK's new `PackageVolume.Repair()` detects and corrects these conditions.
+
+`PackageVolume.VerifyIfOk()` checks if the PackageVolume is OK or in need of repair.
```

Comment:
Should this be VerifyIsOk (is, instead of if)?

Reply:
Changed to `CheckPackageVolume()` and `RepairPackageVolume()` e.g.

```
var packageVolume = PackageVolumeManager.FindPackageVolumeByPath(path);
if (!packageVolume.CheckPackageVolume())
{
    Console.WriteLine("Needs repair");
    packageVolume.RepairPackageVolume();
    Console.WriteLine("Repaired");
}
```